dev/com.ibm.ws.microprofile.metrics/src/com/ibm/ws/microprofile/metrics/writer/JSONMetadataWriter.java

```diff
@@ -90,20 +90,38 @@ private JSONObject getJsonFromMetricMetadataMap(Map<String, Metadata> metadataMa
         return jsonObject;
     }
 
+    @FFDCIgnore({ Exception.class })
     private JSONObject getJsonFromObject(Metadata metadata) {
         JSONObject jsonObject = new JSONObject();
         try {
-            jsonObject.put("name", metadata.getName());
-            jsonObject.put("displayName", metadata.getDisplayName());
-            jsonObject.put("description", Tr.formatMessage(tc, locale, metadata.getDescription()));
-            jsonObject.put("type", metadata.getType());
-            jsonObject.put("unit", metadata.getUnit());
+
+            jsonObject.put("name", checkString(metadata.getName()));
+
+            jsonObject.put("displayName", checkString(metadata.getDisplayName()));
+
+            //Check TR.formatMessage for performance impact
+            jsonObject.put("description", Tr.formatMessage(tc, locale, checkString(metadata.getDescription())));
+
+            jsonObject.put("type", checkString(metadata.getType()));
+
+            jsonObject.put("unit", checkString(metadata.getUnit()));
+
             jsonObject.put("tags", getJsonFromMap(metadata.getTags()));
+
         } catch (Exception e) {
+            Tr.event(tc, "JSON metadata issue: " + e.getMessage());
         }
         return jsonObject;
     }
 
+    String checkString(String s) {
```

Comment:
Can this be a private method? Also, can the name be changed to reflect the return value. (eg parseMetadataValue / sanitizeMetadataValue)